### ISSUE TITLE
Refine `TrainMap`

### ISSUE THREAD

**Issue by dannys4** [MeasureTransport/MParT#334]
For rectified polynomials we can take advantage of the structure of the problem in a couple of ways:
- [x] We should take advantage of when we transport to a Normal distribution in our optimization by separating out the optimization objective by component so we solve `d` one dimensional problems instead of one `d` dimensional problem
- [x] In the KL case, we know the optimization objective is convex. Why not use, e.g., Nesterov accelerated gradient descent?

**Comment by dannys4** [comment]
@mparno I'm all for checking off number 1, but number 2 I've realized is going to be a little more tricky just because now we'll have to start talking about adding optimization back into MParT. I'd be happy adding back the Levenberg-Marquadt optimizer (souped up for GPU), or implementing a completely new optimizer a la nesterov, but should we add fields in `TrainOptions` for these? I.e. have something like a `AlgOptions` class that defaults to `DefaultAlg` (which is no options) and then has child classes like `LevenbergMarquadtAlg: public AlgOptions` which has a value of $\lambda$ for the dampener, and perhaps `NesterovAlg: public AlgOptions`, which has values $\varepsilon$ and $\mu$ for the learning rate and, uh, whatever $\mu$ is supposed to mean (artificial mass? it doesn't matter). Otherwise, should we just fix these constants accordingly? Anyway, any thoughts on this would be nice.

**Comment by mparno** [comment]
I think tackling (1) is more important than (2) and we should probably just skip adding our own optimization stuff for now.  I'm not sure about Julia, but it's relatively easy with pytorch right now to use their optimization stack with an MParT model (see [this example](https://github.com/MeasureTransport/MParT-examples/blob/main/examples/python/FromSamples1D_pytorch.py)), so I'm not super motivated to develop our own tools unless we have a use case where we can't leverage another library.

**Comment by dannys4** [comment]
For this, we should probably expose the map objective class, which I haven't done yet.
